core/src/main/java/org/web3j/tx/Contract.java

```diff
@@ -34,17 +34,18 @@
 /**
  * Solidity contract type abstraction for interacting with smart contracts via native Java types.
  */
+@SuppressWarnings("WeakerAccess")
 public abstract class Contract extends ManagedTransaction {
 
     // https://www.reddit.com/r/ethereum/comments/5g8ia6/attention_miners_we_recommend_raising_gas_limit/
     public static final BigInteger GAS_LIMIT = BigInteger.valueOf(4_300_000);
 
-    private final String contractBinary;
-    private String contractAddress;
-    private final BigInteger gasPrice;
-    private final BigInteger gasLimit;
-    private TransactionReceipt transactionReceipt;
-    private Map<String, String> deployedAddresses;
+    protected final String contractBinary;
```

Comment:
What's your thinking with making all of these protected? Only a subset is required for the proposed change.

Reply:
Basically, the approach I see is that if a dev is choosing the extend this (manually) then they should be able to control things as needed without having the reinvent wheels or duplicate instance variables.

For libraries, I'm all about minimizing exposed surface areas (only create methods that are needed, only make things public that need to be) and also all about keeping things flexible for devs that are extending the base classes - default to "protected" for anything that a subclass might reasonably want to control and keep things private if and only if there's no reasonable use case where a subclass should have a say in a variable's value or use.